File: PalindromeNumber/palindromeNumber.java
public class Solution {
    public boolean isPalindrome(int x) {
        int h = 1;
        
        if (x < 0)  return false;
        
        while(x / h >= 10) {
            h = h * 10;
        }
        
        while(x > 0) {
           if( x / h != x % 10 ) return false;
            x = x % h / 10;
            h = h / 100;
        }
        
        return true;
        
        
    }
}

File: src/PalindromeNumber/palindromeNumber.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    var rex = 0, re = x;
    while(x) {
        rex = rex * 10 + x % 10;
        x = parseInt(x / 10);
    }
    return rex == re && rex >= 0;
};